cimgen/languages/modernpython/utils/base.py

```diff
@@ -132,38 +151,217 @@ def cgmes_attributes_in_profile(self, profile: BaseProfile | None) -> dict[str,
+                infos = dict()
+
                 if f not in self.cgmes_attribute_names_in_profile(profile) or shortname in seen_attrs:
-                    # Wrong profile or already found from a parent.
                     continue
-                else:
-                    # Namespace finding
-                    # "class namespace" means the first namespace defined in the inheritance tree.
-                    # This can go up to Base, which will give the default cim NS.
-                    if (extra := getattr(f.default, "json_schema_extra", None)) is None:
-                        # The attribute does not have extra metadata. It might be a custom atttribute
-                        # without it, or a base type (int...).
-                        # Use the class namespace.
-                        namespace = self.namespace
-                    elif (attr_ns := extra.get("namespace", None)) is None:
-                        # The attribute has some extras, but not namespace.
-                        # Use the class namespace.
-                        namespace = self.namespace
-                    else:
+
+                # Namespace finding
+                # "class namespace" means the first namespace defined in the inheritance tree.
+                # This can go up to Base, which will give the default cim NS.
+                infos["namespace"] = self.namespace
+                extra = getattr(f.default, "json_schema_extra", None)
+                if extra and extra.get("is_used"):
+                    # adding the extras, used for xml generation
+                    extra_info = {
+                        "attr_name": qualname,
+                        "is_class_attribute": extra.get("is_class_attribute"),
+                        "is_enum_attribute": extra.get("is_enum_attribute"),
+                        "is_list_attribute": extra.get("is_list_attribute"),
+                        "is_primitive_attribute": extra.get("is_primitive_attribute"),
+                        "is_datatype_attribute": extra.get("is_datatype_attribute"),
+                        "attribute_class": extra.get("attribute_class"),
+                        "attribute_main_profile": self.get_attribute_main_profile(shortname),
+                    }
+                    if extra.get("namespace"):
                         # The attribute has an explicit namesapce
-                        namespace = attr_ns
+                        extra_info["namespace"] = extra.get("namespace", self.namespace)
+                    infos.update(extra_info)
 
-                    qual_attrs[qualname] = CgmesAttribute(
-                        value=getattr(self, shortname),
-                        namespace=namespace,
-                    )
-                    seen_attrs.add(shortname)
+                infos["value"] = getattr(self, shortname)
+
+                qual_attrs[qualname] = CgmesAttribute(infos)
```

Comment:
```suggestion
                qual_attrs[qualname] = CgmesAttribute(infos)  # type: ignore
```

I don't know if this could be changed that pyright says the types are all correct, that's why the ignore marker.
Additionally I added this to CgmesAttribute:
```
    attr_name: str
    is_class_attribute: bool
    is_enum_attribute: bool
    is_list_attribute: bool
    is_primitive_attribute: bool
    is_datatype_attribute: bool
    attribute_class: str
    attribute_main_profile: BaseProfile
```
Maybe TypedDict is not a good choice as type of CgmesAttribute?